src/condvar.rs

```diff
@@ -383,6 +395,134 @@ impl Condvar {
         let deadline = util::to_deadline(timeout);
         self.wait_until_internal(unsafe { MutexGuard::mutex(mutex_guard).raw() }, deadline)
     }
+
+    #[inline]
```

Comment:
We may want benchmarks for this, but hinting for inlining seems like a good idea.

I don't think there's a reasonable way to reduce the monomorphization cost like `wait_until_internal` since `condition` is inherently generic.

Reply:
I agree, an `#[inline]` hint would be useful here.